test/e2e/resize_nodes.go

```diff
@@ -259,27 +266,46 @@ func resizeRC(c clientset.Interface, ns, name string, replicas int32) error {
 	return err
 }
 
-func getMaster(c clientset.Interface) string {
-	master := ""
+// getMaster populates the externalIP, internalIP and hostname fields of the master.
+// If any of these is unavailable, it is set to "".
+func getMaster(c clientset.Interface) Address {
+	master := Address{}
+
+	// Populate the internal IP.
+	eps, err := c.Core().Endpoints(api.NamespaceDefault).Get("kubernetes")
+	if err != nil {
+		framework.Failf("Failed to get kubernetes endpoints: %v", err)
+	}
+	if len(eps.Subsets) != 1 || len(eps.Subsets[0].Addresses) != 1 {
+		framework.Failf("There are more than 1 endpoints for kubernetes service: %+v", eps)
+	}
+	master.internalIP = eps.Subsets[0].Addresses[0].IP
```

Comment:
You can also just add (since you're re-uping anyway i guess)
```go 
	internalIPs := []string{"10.", "172.", "192.168."}
	for _, cidrs := range internalIPs {
		if strings.HasPrefix(ip, cidrs) {
			fmt.Printf("internal %v\n", ip)
		}
	}
```

slightly inaccurate because the 172 is a /12, but meh